execution/executor/src/lib.rs

```diff
@@ -381,19 +383,48 @@ where
         )?;
 
         info!("Skipping the first {} transactions.", num_txns_to_skip);
+        let txn_list_is_empty = txn_list_with_proof.is_empty();
         let transactions: Vec<_> = txn_list_with_proof
             .transactions
             .into_iter()
             .skip(num_txns_to_skip as usize)
             .collect();
 
         // If the proof is verified, then the length of txn_infos and txns must be the same.
-        let transaction_infos: Vec<_> = txn_list_with_proof
-            .proof
-            .transaction_infos()
-            .iter()
-            .skip(num_txns_to_skip as usize)
-            .collect();
+        let (skipped_transaction_infos, transaction_infos) = {
+            let (a, b) = txn_list_with_proof
+                .proof
+                .transaction_infos()
+                .split_at(num_txns_to_skip as usize);
+            (a.to_vec(), b.to_vec())
+        };
+
+        // verify no fork happens.
+        if !txn_list_is_empty {
+            let frozen_subtree_roots_from_proof = txn_list_with_proof
```

Comment:
add comment, like

"# Left side of the proof happens to be the frozen subtree roots of the accumulator right before the list of txns are applied."